docs/content/envoy-external-authorization.md

```diff
@@ -0,0 +1,334 @@
+    perm.path = http_request.path
+}
+
+user_name = parsed {
+    [_, encoded] := split(http_request.headers.authorization, " ")
+    [parsed, _] := split(base64url.decode(encoded), ":")
+}
+
+user_roles = {
+    "alice": ["guest"],
+    "bob":   ["admin"]
+}
+
+role_perms = {
+    "guest": [
+        {"method": "GET",  "path": "/people"},
+    ],
+    "admin": [
+        {"method": "POST",  "path": "/people"},
+        {"method": "GET",   "path": "/people"},
+    ],
+}
+```
+
+```bash
+kubectl create configmap opa-policy --from-file policy.rego
+```
+
+An example of the input received by OPA can be seen [here](https://github.com/open-policy-agent/opa-istio-plugin#example-input).
+
+> In typical deployments the policy would either be built into the OPA container
+> image or it would fetched dynamically via the [Bundle
+> API](https://www.openpolicyagent.org/docs/latest/bundles/). ConfigMaps are
+> used in this tutorial for test purposes.
+
+### 4. Create App Deployment with OPA and Envoy sidecars
+
+Our deployment contains a sample Go app which provides information about employees in a company. It exposes a `/people` endpoint to `get` and `create` employees. More information can on the app be found [here](https://github.com/ashutosh-narkar/go-test-server).
+
+Save the deployment as **deployment.yaml**:
+
+```yaml
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: example-app
+  labels:
+    app: example-app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: example-app
+  template:
+    metadata:
+      labels:
+        app: example-app
+    spec:
+      initContainers:
+        - name: proxy-init
+          image: openpolicyagent/proxy_init:v2
+          args: ["-p", "8000", "-u", "1111"]
+          securityContext:
+            capabilities:
+              add:
+              - NET_ADMIN
+            runAsNonRoot: false
+            runAsUser: 0
+      containers:
+        - name: app
+          image: openpolicyagent/demo-test-server:v1
+          ports:
+            - containerPort: 8080
+        - name: envoy
+          image: envoyproxy/envoy:v1.10.0
+          securityContext:
+            runAsUser: 1111
+          volumeMounts:
+          - readOnly: true
+            mountPath: /config
+            name: proxy-config
+          args:
+          - "envoy"
+          - "--config-path"
+          - "/config/envoy.yaml"
+        - name: opa
+          # Note: openpolicyagent/opa:edge-istio is created by retagging
+          # the latest released image of OPA-Istio.
+          image: openpolicyagent/opa:{{< current_opa_istio_docker_version >}}
+          securityContext:
+            runAsUser: 1111
+          volumeMounts:
+          - readOnly: true
+            mountPath: /config
+            name: opa-config
+          - readOnly: true
+            mountPath: /policy
+            name: opa-policy
+          args:
+          - "--plugin-dir=/app"
+          - "run" 
+          - "--server"
```

Comment:
Let's turn on the console decision logger here. Once #1554 is fixed we'll need that anyway.